github_access/templates/github_access/access_request_form.html

```diff
@@ -1,110 +1,256 @@
+      $('#github-repo-selection-empty').hide();
+      $('#github-repo-selection-list').show();
+      $('#github-repo-selection-list').children('nav').css('display', 'flex');
+      $('#github-repo-selected-count').show();
+      $('#github-repo-selected-count').text(finalCount + ' selected');
     }
-    function repoListChange() {
-      labelComponents = JSON.parse($('#github_accessLabel').val());
-      if(!labelComponents.length){
-        labelComponents = [{}]
-      }
-      labelComponents[0]["repoList"] = getSelectedRepos()
-      $('#github_accessLabel').val(JSON.stringify(labelComponents));
+  };
+
+  const selectGithubRepoCheckbox = (elem, checked) => {
+    if(checked) {
+      $(elem).find('div').show();
+      $(elem).find('input').prop('checked', true);
+      $(elem).removeClass('hover:bg-blue-50 hover:text-blue-700').addClass('bg-gray-50');
+      $(elem).children('td#description-td').removeClass('text-gray-900').addClass('text-blue-600');
+
+    } else {
+      $(elem).find('div').hide();
+      $(elem).find('input').prop('checked', false);
+      $(elem).removeClass('bg-gray-50').addClass('hover:bg-blue-50 hover:text-blue-700');
+      $(elem).children('td#description-td').removeClass('text-blue-600').addClass('text-gray-900');
     }
-    function githubAccessLevelChange() {
-      labelComponents = JSON.parse($('#github_accessLabel').val());
-      if (!labelComponents.length) {
-        labelComponents = [{}]
+  };
+
+  const addGithubReposelection = (elem) => {
+    const selectionList = $('#github-repo-selection-table');
+    const newSpan = $("#github-repo-selection-row-template").clone(true, true);
+
+    selectGithubRepoCheckbox(elem, true);
+    const org_name = $(elem).attr("org_name");
+    const repo_name = $(elem).attr("repo_name");
+    const repo_id = $(elem).attr("repo_id");
+
+    newSpan.appendTo(selectionList);
+    newSpan.show();
+    const tableData = newSpan.children('td');
+
+    tableData[0].textContent = org_name + "/" + repo_name;
+    newSpan.attr('org_name', org_name);
+    newSpan.attr('repo_name', repo_name);
+    newSpan.attr('repo_id', repo_id)
+    selectedList[repo_name] = true;
+
+    handleGithubReposelectionView();
+  };
+
+  const removeRepoSelectionSpanElem = (rightElem, leftElem) => {
+    rightElem.remove();
+    selectGithubRepoCheckbox(leftElem, false)
+
+    selectedList[$(leftElem).attr("repo_id") || $(rightElem).attr("repo_id")] = false;
+    handleGithubReposelectionView();
+  };
+
+  const removeAllGithubRepos = () => {
+    const repos = $('#github-repo-selection-table').children('tr');
+
+    for(iter = 0; iter < repos.length; iter++) {
+      removeRepoSelectionSpanElem(repos[iter], $("#github-repo-table").find(`tr[repo_id="${$(repos[iter]).attr('repo_id')}"]`));
+    }
+    $("#selectAllGithubRepo").prop("checked", false);
+    updateSelectedGithubRepo();
+  };
+
+  const selectAllGithubRepo = () => {
+    const repos = $('#github-repo-table').children('tr');
+
```

Comment:
need to call updateSelectedGithubRepo() here as well

Reply:
Done.